cosmos-tx/tests/integration.rs

```diff
@@ -0,0 +1,126 @@
+//! Integration test which submits transactions to a local `gaia` node.
+//!
+//! Requires Docker.
+
+#![cfg(feature = "rpc")]
+
+use cosmos_tx::{bank::MsgSend, rpc, AccountId, Builder, Coin, Fee, MsgType, SigningKey};
+use std::{panic, process, str, thread, time::Duration};
+
+/// Name of the Docker image (on Docker Hub) to use
+const DOCKER_IMAGE: &str = "jackzampolin/gaiatest";
+
+/// Chain ID to use for tests
+const CHAIN_ID: &str = "cosmos-tx-test";
+
+/// RPC port
+const RPC_PORT: u16 = 26657;
+
+/// Denom name
+const DENOM: &str = "uatom";
+
+/// Expected account number
+const ACCOUNT_NUMBER: u64 = 0;
+
+/// Example memo
+const EXAMPLE_MEMO: &str = "test memo";
+
+#[test]
+fn msg_send() {
+    let sender_private_key = SigningKey::random();
+    let recipient_private_key = SigningKey::random();
+
+    let amount = Coin {
+        amount: 1_000_000.into(),
+        denom: DENOM.parse().unwrap(),
+    };
+
+    let msg_send = MsgSend {
+        from_address: sender_private_key.account_id(),
+        to_address: recipient_private_key.account_id(),
+        amount: vec![amount.clone()],
+    };
+
+    let chain_id: tendermint::chain::Id = CHAIN_ID.parse().unwrap();
+    let sequence_number = 0;
+    let gas = 100_000;
+    let fee = Fee::from_amount_and_gas(amount, gas);
+    let timeout_height = 9001u16;
+
+    let tx = Builder::new(chain_id, ACCOUNT_NUMBER)
+        .sign_tx(
+            &sender_private_key,
+            sequence_number,
+            &[msg_send.to_msg().unwrap()],
+            fee,
+            EXAMPLE_MEMO,
+            timeout_height.into(),
+        )
+        .unwrap();
+
+    run_docker(sender_private_key.account_id(), || {
+        // TODO(tarcieri): poll for readiness instead of blindly waiting
+        thread::sleep(Duration::from_secs(2));
```

Comment:
Might be nice to have an upstream solution for this in `tendermint-rpc` (i.e. polling for readiness within a timeout waiting for a node to boot)

cc @thanethomson

Reply:
Upstreamed this and it was released in `tendermint-rpc` v0.19 🎉